.github/workflows/dockerhub.yml

```diff
@@ -1,228 +1,311 @@
 name: Docker Hub build
+
 on:
   push:
     branches:
-      - "develop"
-      - "master"
+      - develop
+      - master
       - "release/*"
+    tags:
+      - "carma-system-*"
 
 jobs:
+  determine_docker_org_and_tag:
+    runs-on: ubuntu-latest
+    outputs:
+      docker_organization: ${{ steps.docker-org-and-tag.outputs.docker_organization }}
+      docker_image_tag: ${{ steps.docker-org-and-tag.outputs.docker_image_tag }}
+    steps:
+      - id: docker-org-and-tag
+        uses: usdot-fhwa-stol/actions/docker-org-and-tag@main
+
   dockerhub-telematic-cloud-messaging:
+    needs: determine_docker_org_and_tag
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+      - name: Get Docker Image Metadata
+        shell: bash
+        id: docker-image-metadata
+        run: |
+          echo build_date=$(date -u +”%Y-%m-%dT%H:%M:%SZ”) >> $GITHUB_OUTPUT
+          echo vcs_ref=$(git rev-parse --short HEAD) >> $GITHUB_OUTPUT
```

Comment:
What does this vcs_ref mean?

Reply:
This git commit reference we are capturing it from actions repo usually contains a short reference to the specific git commit https://github.com/usdot-fhwa-stol/actions/blob/main/docker-build-metadata/action.yml